### PR TITLE
Jbaker/simpler icon template/#48

### PR DESCRIPTION
**Pull request by jason-capsule42**
# Alaska Airlines Pull Request

**Fixes:** #36

## Summary:

This change implements an alternative templating solution provided by @geoffrich here: #48 (comment).

## Type of change:

Please delete options that are not relevant.

- [ ] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

**Review by blackfalcon — APPROVED** [review]

**Review by blackfalcon — CHANGES_REQUESTED** [review]
This is a great update! Please update the commit message to be `refactor`. 

This update is a refactor of the base code but has no impact on the released package.

File: src/auro-input.js [hunk 1]
```diff
@@ -161,6 +162,7 @@ export default class AuroInput extends BaseInput {
 
     if (this.icon) {
       this.inputIconName = card.cardIcon;
+      console.warn(this.inputIconName);
```
Comment (geoffrich) [inline src/auro-input.js:165]: Remove console log

File: src/auro-input.js [hunk 2]
```diff
@@ -6,6 +6,7 @@
 /* eslint-disable max-lines */
 
 import { html } from "lit-element";
+import {repeat} from 'lit-html/directives/repeat.js';
```
Comment (geoffrich) [inline src/auro-input.js:9]: For consistency
```suggestion
import { repeat } from 'lit-html/directives/repeat.js';
```

File: src/auro-input.js [hunk 3]
```diff
@@ -281,14 +283,15 @@ export default class AuroInput extends BaseInput {
...
-        ${this.inputIconName === 'cc-amex' ? html`<auro-icon class="creditCard-icon" category="payment" name="cc-amex"></auro-icon>` : undefined}
-        ${this.inputIconName === 'cc-visa' ? html`<auro-icon class="creditCard-icon" category="payment" name="cc-visa"></auro-icon>` : undefined}
-        ${this.inputIconName === 'cc-mastercard' ? html`<auro-icon class="creditCard-icon" category="payment" name="cc-mastercard"></auro-icon>` : undefined}
-        ${this.inputIconName === 'cc-discover' ? html`<auro-icon class="creditCard-icon" category="payment" name="cc-discover"></auro-icon>` : undefined}
-        ${this.inputIconName === 'cc-alaska' ? html`<auro-icon class="creditCard-icon" category="payment" name="cc-alaska"></auro-icon>` : undefined}
+        `
+      }
+      ${this.type === 'credit-card' ? this.processCreditCard() : undefined}
+      ${this.inputIconName ?
+        repeat([this.inputIconName], x => x, (name) =>
```
Comment (geoffrich) [inline src/auro-input.js:290]: I think it's worth a comment here detailing what we're trying to accomplish.

File: src/auro-input.js (hunk 1 above)
Comment (blackfalcon) [reply src/auro-input.js:165]: Opps, totally missed that.

File: src/auro-input.js (hunk 3 above)
Comment (blackfalcon) [reply src/auro-input.js:290]: 100. Even though I am approving this update to make this code less cumbersome, I don't want to lose sight of the real issue that auro-icon isn't updating the fetch call with the attribute update. https://github.com/AlaskaAirlines/auro-icon/issues/31

**Review by blackfalcon — CHANGES_REQUESTED** [review]

**Review by blackfalcon — APPROVED** [review]

**Pull request merged**